File: testdata/repositories/jsts/src/http-registrations.ts
import express, {type Request, type Response} from "express"

const application = express()

export function emptyHealthHandler(_request: Request, _response: Response): void {}

export function returnedHealthHandler(): typeof emptyHealthHandler {
  function emptyReturnedHandler(_request: Request, _response: Response): void {}
  return emptyReturnedHandler
}

application.get("/health", emptyHealthHandler)
application.get("/returned", returnedHealthHandler())

class RouteDefinition {
  constructor(public path: string) {}
}

function registerRoute(path: string): void {
  application.get(path, emptyHealthHandler)
}

export function installRoutes(runtimePath: string): void {
  const update = new RouteDefinition("/v1/update")
  const metrics = new RouteDefinition("/v1/metrics")
  registerRoute(update.path)
  registerRoute(metrics.path)
  registerRoute(runtimePath)
}

export function unusedRouteDefinition(): RouteDefinition {
  return new RouteDefinition("/never-registered")
}

const localRouteNames = { get(_path: string): void {} }
localRouteNames.get("/not-a-route")

class MethodArgumentApplication {
  first(next: unknown): unknown { return next }
  second(next: unknown): unknown { return next }
}

function receiveMethodArguments(...values: Array<(next: unknown) => unknown>): void {}

// JS method values are passed functions; no bound receiver execution is implied.
export function passMethodArguments(app: MethodArgumentApplication): void {
  receiveMethodArguments(app.first, app.second, app.first, (app.second))
}

File: testdata/repositories/python/src/fixture_app/http_registrations.py
from fastapi import FastAPI

application = FastAPI()


@application.get("/health")
def empty_health_handler():
    pass


class RouteDefinition:
    def __init__(self, path):
        self.path = path


def register_route(path):
    @application.get(path)
    def empty_registered_handler():
        pass
    return empty_registered_handler


def install_routes(runtime_path):
    update = RouteDefinition("/v1/update")
    metrics = RouteDefinition("/v1/metrics")
    register_route(update.path)
    register_route(metrics.path)
    register_route(runtime_path)


def unused_route_definition():
    return RouteDefinition("/never-registered")


class LocalRouteNames:
    def get(self, path):
        return empty_health_handler


def local_name_control():
    return LocalRouteNames().get("/not-a-route")


class MethodArgumentApplication:
    def first(self, next_handler):
        return next_handler

    def second(self, next_handler):
        return next_handler


def receive_method_arguments(*values):
    pass


def pass_method_arguments(app: MethodArgumentApplication):
    receive_method_arguments(app.first, app.second, app.first, (app.second))
